win32ss/user/ntuser/nonclient.c

```diff
@@ -494,41 +473,66 @@ DefWndDoSizeMove(PWND pwnd, WORD wParam)
              UserDrawMovingFrame( hdc, &sizingRect, thickframe );
 	  }
 
-	  if (msg.message == WM_KEYDOWN) UserSetCursorPos(pt.x, pt.y, 0, 0, FALSE);
-	  else
-	  {
-	      RECT newRect = unmodRect;
+      if (msg.message == WM_KEYDOWN)
+      {
+          UserSetCursorPos(pt.x, pt.y, 0, 0, FALSE);
+      }
+      else
+      {
+          RECT newRect = unmodRect;
 
-	      if (!iconic && !DragFullWindows) UserDrawMovingFrame( hdc, &sizingRect, thickframe );
+          if (!iconic && !DragFullWindows)
+          {
+              UserDrawMovingFrame(hdc, frameRect, thickframe);
+              frameRect = &sizingRect;
+          }
           if (hittest == HTCAPTION)
           {
               /* Restore window size if it is snapped */
-              if (!RECTL_bIsEmptyRect(&pwnd->InternalPos.NormalRect) &&
-                  !IntEqualRect(&pwnd->InternalPos.NormalRect, &pwnd->rcWindow))
+              RECT *r = &newRect;
+              LONG width, height, capcy, snapTo;
+              if (snap && syscommand == SC_MOVE && !iconic &&
+                  !RECTL_bIsEmptyRect(&pwnd->InternalPos.NormalRect))
               {
-                  UserSetCursorPos(max(0, pwnd->InternalPos.NormalRect.left) + pt.x, pwnd->InternalPos.NormalRect.top + pt.y, 0, 0, FALSE);
-
-                  /* Save normal size - it required when window unsnapped from one side and snapped to another holding mouse down */
-                  origRect = pwnd->InternalPos.NormalRect;
-
-                  /* Restore from maximized state */
-                  if (Style & WS_MAXIMIZE)
+                  *r = pwnd->InternalPos.NormalRect;
+                  origRect = *r; /* Save normal size - it required when window unsnapped from one side and snapped to another holding mouse down */
```

Comment:
```suggestion
                  origRect = *r; /* Save normal size - is required when window unsnapped from one side and snapped to another holding mouse down */
```

Reply:
Not my typo, I just preserved it 😂

Reply:
you are hereby officially allowed to fix it. ;-)